contracts/PercentageTransferManager/Cargo.toml

```diff
@@ -38,6 +38,10 @@ std = [
     "scale-info",
     "scale-info/std",
 ]
+
+[profile.release]
+overflow-checks = false
```

Comment:
I can see how wrapping is right in mainnet.
But are we OK with silent wrapping arithmetic in a testnet?
Panicking might help us catch bugs in the interim.

Reply:
Can you ELI5 this flag please?

Reply:
This flag is introduced to overcome the rustc bug related to wasm. 

`overflow-checks` are checks inserted by the Rust compiler for integer arithmetic. those do panic. if there was no bug we would have direct aborts instead of panics in our Wasm builds, but due to this bug not all of those get filtered out leaving us with those strange errors {Simillar Remon faced today} not inserting those overflow checks in the first place simply removes some category of potential places where the Rust compiler can fail.
in Solidity speak: overflow-checks is similar to safe-math`

Reply:
If I remember correctly, with the flag set to true, arithmetic over or underflow will result in panics. When set to false, it will instead result in wrapping arithmetic. So `0 - 1 == usize::MAX` for example

Reply:
Why was an overflow being triggered in the first place? I don't think disabling overflow checks is the right solution.

Reply:
So there is a statement in the contract i.e `10u128.pow(6)` which is a valid statement if we compile it into rust but during wasm compilation, it panics and colludes the wasm blob. It is a bug in the compiler so make it work I switch off the `overflow-checks` but another way we can achieve this by directly using `1000000` directly without using the `pow` math operation.

Reply:
Let's directly use `1_000_000` and report this upstream.

Reply:
It is already a well-known issue and Robin suggested to use `overflow-checks`

Reply:
But anyway I am fine with `1_000_000` this also

Reply:
Hmm.. This will also not work as well I thought it will work but it isn't. Somehow there is a problem in the arithmetic operation to make it compile in wasm

Reply:
Then this pow thing wasn't the issue. It being a known problem does not make it safe. We need to find the actual loc that is causing this issue.